example/extensions/lib_custom_op/relu_lib.cc

```diff
@@ -0,0 +1,171 @@
+
+MyStatefulReluCPU::MyStatefulReluCPU(const std::unordered_map<std::string, std::string>& attrs)
+  : attrs_(attrs) {}
+
+MXReturnValue MyStatefulReluCPU::Forward(std::vector<MXTensor>* inputs,
+                                         std::vector<MXTensor>* outputs,
+                                         const OpResource& op_res) {
+  return forwardCPU(attrs_, inputs, outputs, op_res);
+}
+
+MXReturnValue MyStatefulReluCPU::Backward(std::vector<MXTensor>* inputs,
+                                          std::vector<MXTensor>* outputs,
+                                          const OpResource& op_res) {
+  return backwardCPU(attrs_, inputs, outputs, op_res);
+}
+
+MyStatefulReluGPU::MyStatefulReluGPU(const std::unordered_map<std::string, std::string>& attrs)
+  : attrs_(attrs) {}
+
+MXReturnValue MyStatefulReluGPU::Forward(std::vector<MXTensor>* inputs,
+                                         std::vector<MXTensor>* outputs,
+                                         const OpResource& op_res) {
+  return forwardGPU(attrs_, inputs, outputs, op_res);
+}
+
+MXReturnValue MyStatefulReluGPU::Backward(std::vector<MXTensor>* inputs,
+                                          std::vector<MXTensor>* outputs,
+                                          const OpResource& op_res) {
+  return backwardGPU(attrs_, inputs, outputs, op_res);
+}
+
+
+MXReturnValue createOpStateCPU(const std::unordered_map<std::string, std::string>& attrs,
+                               CustomStatefulOp** op_inst) {
+  *op_inst = new MyStatefulReluCPU(attrs);
+  return MX_SUCCESS;
+}
+
+MXReturnValue createOpStateGPU(const std::unordered_map<std::string, std::string>& attrs,
+                               CustomStatefulOp** op_inst) {
+  *op_inst = new MyStatefulReluGPU(attrs);
+  return MX_SUCCESS;
+}
+
+REGISTER_OP(my_state_relu)
+.setParseAttrs(parseAttrs)
+.setInferType(inferType)
+.setInferShape(inferShape)
+.setCreateOpState(createOpStateCPU, "cpu")
+.setCreateOpState(createOpStateGPU, "gpu");
+
+MXReturnValue noisyForwardCPU(const std::unordered_map<std::string, std::string>& attrs,
+                              std::vector<MXTensor>* inputs,
+                              std::vector<MXTensor>* outputs,
+                              const OpResource& res) {
+  float* in_data = inputs->at(0).data<float>();
+  float* out_data = outputs->at(0).data<float>();
+
+  mx_cpu_rand_t* states = res.get_cpu_rand_states();
+  std::normal_distribution<float> dist_normal;
+
+  for (int i=0; i<inputs->at(0).size(); ++i) {
+    float noise = dist_normal(*states);
+    out_data[i] = in_data[i] + noise > 0 ? in_data[i] + noise : 0;
+  }
+  return MX_SUCCESS;
+}
+
+REGISTER_OP(my_noisy_relu)
+.setParseAttrs(parseAttrs)
+.setInferType(inferType)
+.setInferShape(inferShape)
+.setForward(noisyForwardCPU, "cpu")
+.setForward(noisyForwardGPU, "gpu")
+.setBackward(backwardCPU, "cpu")
+.setBackward(backwardGPU, "gpu");
+
+MXReturnValue initialize(int version) {
+  if (version >= 20000) {
```

Comment:
are you sure about this? since gemm_lib is still 10700

Reply:
Yes, we should update all the examples to 20000 on master. ill do that in the next PR.

Reply:
discussed offline and we will change example corresponding to master